src/shaders/blur.h

```diff
@@ -1,18 +1,25 @@
 
-vec4 textureCond(sampler2D layerSampler, vec2 pos, bool unnormalized) {
+vec4 textureCond(sampler2D layerSampler, uint layerIdx, vec2 pos, bool unnormalized) {
     vec2 texSize = textureSize(layerSampler, 0);
+    vec2 coord = pos;
 #ifndef BLUR_DONT_SCALE
-    pos *= u_scale[0];
+    coord = ((coord + u_offset[layerIdx]) * u_scale[layerIdx]);
+
+    if (coord.x < 0.0f       || coord.y < 0.0f ||
+        coord.x >= texSize.x || coord.y >= texSize.y) {
+        float border = (u_borderMask & (1u << layerIdx)) != 0 ? 1.0f : 0.0f;
+        return vec4(0.0f, 0.0f, 0.0f, border);
+    }
```

Comment:
Fwiw, I originally didn't do this because of performance concerns (this function is called a lot with a high radius), but if that isn't an issue in practice it's okay I guess.

Reply:
Should be fine. Doing it this way also blurs the edges which is cool.